src/HonzaBotner.Discord.Services/VoiceManager.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Timers;
+using DSharpPlus;
+using DSharpPlus.Entities;
+using DSharpPlus.EventArgs;
+using Timer = System.Threading.Timer;
+
+namespace HonzaBotner.Discord.Services
+{
+    public class VoiceManager : IVoiceManager
+    {
+        private readonly IGuildProvider _guildProvider;
+        private readonly DiscordWrapper _discordWrapper;
+        private DiscordClient Client => _discordWrapper.Client;
+
+        public VoiceManager(DiscordWrapper discordWrapper, IGuildProvider guildProvider)
+        {
+            _discordWrapper = discordWrapper;
+            _guildProvider = guildProvider;
+        }
+
+        public Task Run()
+        {
+            Client.VoiceStateUpdated += Client_VoiceStateUpdated;
+
+            return Task.CompletedTask;

```

Comment:
Will this work? Shouldnt this instead return infinite task?

Reply:
It works, but dunno if it is OK.